### PR TITLE
Add support for intersection type

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
$subject.
Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2619

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java [hunk 1]
```diff
@@ -94,7 +94,8 @@ public static void validateOnBinaryMessageFunction(FunctionTypeSymbol functionTy
             for (ParameterSymbol inputParam : inputParams) {
                 String moduleId = getModuleId(inputParam);
                 String paramSignature = inputParam.typeDescriptor().signature();
-                if (!paramSignature.equals(BYTE_ARRAY) && !paramSignature.equals(moduleId + COLON + CALLER)) {
+                if (!paramSignature.equals(BYTE_ARRAY) && !paramSignature.equals(moduleId + COLON + CALLER) &&
+                        !(inputParam.typeDescriptor().typeKind() == TypeDescKind.INTERSECTION)) {
```
Comment (shafreenAnfar) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java:98]: Does this mean going forward we only support `readonly & byte[] data` ?

Comment (Bhashinee) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java:98]: No, this only says it doesn't give any compiler plugin error when we have `readonly & byte[] data`. It supports both `byte[]` and `readonly & byte[] data`.

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_40/service.bal [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+   }
+}
+
+service isolated class WsService {
+    *websocket:Service;
+
+    remote function onBinaryMessage(websocket:Caller caller, readonly & byte[] data) {
+    }
+
+    remote function onPing(websocket:Caller caller, readonly & byte[] data) {
```
[the remark this answers is not in the document]
Comment (Bhashinee) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_40/service.bal:15]: yes, the caller is optional.

File: compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java (hunk 1 above)
Comment (shafreenAnfar) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java:98]: Nice!

File: compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_40/service.bal (hunk 2 above)
[the remark this answers is not in the document]
Comment (shafreenAnfar) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_40/service.bal:15]: Great!

**Review by Maninda — APPROVED** [review]

**Pull request merged**